edgelet/edgelet-docker/src/migc_persistence.rs

```diff
@@ -0,0 +1,294 @@
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+use std::time::UNIX_EPOCH;
+use std::{collections::HashMap, fs, time::Duration};
+
+use edgelet_core::Module;
+use edgelet_settings::base::image::MIGCSettings;
+
+use crate::{DockerModule, Error};
+
+#[derive(Debug, Clone)]
+struct MIGCPersistenceInner {
+    filename: String,
+    settings: Option<MIGCSettings>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MIGCPersistence {
+    inner: Arc<Mutex<MIGCPersistenceInner>>,
+}
+
+impl MIGCPersistence {
+    pub fn new(filename: String, settings: Option<MIGCSettings>) -> Self {
+        Self {
+            inner: Arc::new(Mutex::new(MIGCPersistenceInner { filename, settings })),
+        }
+    }
+
+    pub async fn record_image_use_timestamp(
+        &self,
+        name_or_id: &str,
+        is_image_id: bool,
+        image_name_to_id: HashMap<String, String>, // HashMap<image_name, image_id>
+    ) {
+        if is_image_id {
+            let id: String = name_or_id.chars().skip(7).take(12).collect();
```

Comment:
Why skip 7 take 12. Recommend comment.

Reply:
// a call to underlying docker API /images/json returns the image list with
            // the ID of each image in the form: "sha256:e216a057b1cb1efc11f8a268f37ef62083e70b1b38323ba252e25ac88904a7e8"
            // The following code skips the "sha256:" part, and takes the first 12 characters
            // (to match with the image id we see when we run 'docker images')